libnavigation-base/src/test/java/com/mapbox/navigation/base/internal/route/AnnotationsRefresherTest.kt

```diff
@@ -0,0 +1,304 @@
+
+@RunWith(Parameterized::class)
+class AnnotationsRefresherTest(
+    private val oldAnnotation: LegAnnotation?,
+    private val newAnnotation: LegAnnotation?,
+    private val legGeometryIndex: Int,
+    private val expectedMergedAnnotation: LegAnnotation?,
+    private val description: String,
+) {
+
+    companion object {
+
+        @JvmStatic
+        @Parameterized.Parameters(name = "{4}")
+        fun data(): Collection<Array<Any?>> {
+            val defaultAnnotation = LegAnnotation.builder().build()
+            val annotationWithCongestionNumericOnlyEmpty = defaultAnnotation.toBuilder()
+                .congestionNumeric(listOf())
+                .build()
+            val annotationWithCongestionNumericOnlyFilled = defaultAnnotation.toBuilder()
+                .congestionNumeric(listOf(1, 2, 3, 4, 5))
+                .build()
+            return listOf(
+                arrayOf(
+                    null,
+                    null,
+                    0,
+                    null,
+                    "Everything is null"
+                ),
+                arrayOf(
+                    defaultAnnotation,
+                    null,
+                    0,
+                    null,
+                    "New annotation is null"
+                ),
+                arrayOf(
+                    defaultAnnotation,
+                    defaultAnnotation,
+                    0,
+                    defaultAnnotation,
+                    "2 default annotations, index = 0"
+                ),
+                arrayOf(
+                    defaultAnnotation,
+                    defaultAnnotation,
+                    5,
+                    defaultAnnotation,
+                    "2 default annotations, index = 5"
+                ),
+                arrayOf(
+                    annotationWithCongestionNumericOnlyEmpty,
+                    defaultAnnotation,
+                    0,
+                    defaultAnnotation,
+                    "Empty congestion_numeric + default, index = 0"
+                ),
+                arrayOf(
+                    annotationWithCongestionNumericOnlyEmpty,
+                    defaultAnnotation,
+                    0,
+                    defaultAnnotation,
+                    "Empty congestion_numeric + default, index = 5"
+                ),
+                arrayOf(
+                    defaultAnnotation,
+                    annotationWithCongestionNumericOnlyEmpty,
+                    0,
+                    annotationWithCongestionNumericOnlyEmpty,
+                    "Default + empty congestion_numeric, index = 0"
+                ),
+                arrayOf(
+                    defaultAnnotation,
+                    annotationWithCongestionNumericOnlyEmpty,
+                    5,
+                    LegAnnotation.builder().congestionNumeric(listOf(0, 0, 0, 0, 0)).build(),
+                    "Default + empty congestion_numeric, index = 5"
```

Comment:
I need a few minutes for each case to understand why it's needed. Usually a test name explains the purpose, maybe it's worth doing the same in a parameter description?
```suggestion
                    "Default + empty congestion_numeric, index = 5. Annotations before current geometry index should be updated to default value"
```

Reply:
The expected value is on the previous line. I don't think we need to include everything in the description. Here its purpose is mostly to simplify finding a case which failed (you run all the cases, one fails, how do you find which one?).

Reply:
> The expected value is on the previous line.

It's not the problem I experience. It's not very hard to find an expected value for me. BTW it's not as easy as it could be because I need to remember the parameters order. From time to time I scroll up to definition to check what is the third parameter for example.

The issue I have is to understand purpose of a test case. When I read this tests I have to compare input/output to understand what is the difference and come up with an idea of what this case is about. It would be easier for me to read if test cases contained an explanation of why they are needed. 

> I don't think we need to include everything in the description. Here its purpose is mostly to simplify finding a case which failed

I'm okay if you add explanation to each test case in a way you like. I would put it to the description, but if you want to implement it differently - I'm fine with it 😉

Reply:
Will add some description.